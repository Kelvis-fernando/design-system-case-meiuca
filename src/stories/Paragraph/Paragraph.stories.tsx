import type { Meta, StoryObj } from "@storybook/react";
import { Paragraph } from "./Paragraph";

const meta = {
  title: "Components/Paragraph",
  component: Paragraph,
  tags: ["autodocs"],
} satisfies Meta<typeof Paragraph>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    text: "Testing paragraph",
  },
};
