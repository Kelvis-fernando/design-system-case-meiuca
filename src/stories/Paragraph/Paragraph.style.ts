import tokens from "../../../tokens/global.json";

export const paragraphStyle: React.CSSProperties = {
  color: tokens.neutral.color[5].value,
  fontFamily: tokens.font.family.highlight.value,
  fontSize: tokens.font.size.xs.value,
};
