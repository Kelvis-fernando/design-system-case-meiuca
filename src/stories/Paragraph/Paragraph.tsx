import { ParagraphProps } from "../../types/Paragraph";
import { paragraphStyle } from "./Paragraph.style";

export const Paragraph: React.FC<ParagraphProps> = ({ text, ...props }) => {
  return (
    <p style={paragraphStyle} {...props}>
      {text}
    </p>
  );
};
