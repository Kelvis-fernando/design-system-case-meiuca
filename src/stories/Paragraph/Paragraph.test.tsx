import { render, screen } from "@testing-library/react";
import { Paragraph } from "./Paragraph";

test("should render Paragraph", () => {
  render(<Paragraph text={"Paragraph"} />);
  const linkElement = screen.getByText(/Paragraph/i);

  expect(linkElement).toBeInTheDocument();
});
