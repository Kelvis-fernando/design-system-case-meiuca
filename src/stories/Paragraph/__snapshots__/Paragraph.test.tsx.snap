// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Paragraph 1`] = `
<div>
  <p
    style="color: rgb(0, 0, 0); font-family: 'Archivo'; font-size: 16px;"
  >
    Paragraph
  </p>
</div>
`;
